src/lib/notifier.c

```diff
@@ -42,6 +44,16 @@ int notifier_register(void *receiver, void *caller, enum notify_id type,
 
 	assert(type >= NOTIFIER_ID_CPU_FREQ && type < NOTIFIER_ID_COUNT);
 
+	/* Find already registered event of this type */
+	if (flags & NOTIFIER_FLAG_AGGREGATE &&
+	    !list_is_empty(&notify->list[type])) {
+		handle = container_of((&notify->list[type])->next,
```

Comment:
notify is local per core, so probably atomic is not needed

Reply:
Done.